core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -337,19 +413,27 @@ private boolean fullRefresh(MatViewRefreshTask refreshTask) {
                 // is used to initialize base table readers returned from the refreshExecutionContext.getReader()
                 // call, so that all of them are at the same txn.
                 engine.detachReader(baseTableReader);
-                refreshExecutionContext.of(baseTableReader);
+                refreshSqlExecutionContext.of(baseTableReader);
                 try {
                     walWriter.truncateSoft();
-                    resetInvalidState(state, walWriter);
+                    resetInvalidState(viewState, walWriter);
```

Comment:
I don't see where `last period hi` is reset in the view invalidation / full refresh workflow, should it be somewhere here or when view is marked as invalid?

Reply:
```java
    private void resetInvalidState(MatViewState viewState, WalWriter walWriter) {
        viewState.markAsValid();
        viewState.setLastRefreshBaseTableTxn(-1);
        viewState.setLastRefreshTimestamp(Numbers.LONG_NULL);
        viewState.setLastPeriodHi(Numbers.LONG_NULL); // <- the reset is here
        // ...
```